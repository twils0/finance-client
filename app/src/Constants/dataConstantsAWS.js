// eslint-disable-next-line
export const actionTypes = {
  SET_AWS_STATUS: 'DATA_AUTH_AWS_STATUS',
  SET_AWS_USER: 'DATA_AUTH_SET_AWS_USER',
  RESET_AWS_STATE: 'DATA_AUTH_RESET_AWS_STATE',
};
