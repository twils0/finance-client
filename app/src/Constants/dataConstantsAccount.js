export const actionTypes = {
  SET_ACCOUNT_STATUS: 'DATA_ACCOUNT_SET_ACCOUNT_STATUS',
  SET_FIELD: 'DATA_ACCOUNT_SET_FIELD',
  SET_FIELDS: 'DATA_ACCOUNT_SET_FIELDS',
  RESET_ACCOUNT_STATE: 'DATA_ACCOUNT_RESET_ACCOUNT_STATE',
};

export const statusNames = {
  AWS_FIELDS: 'accountStatusAWSFields',
  STRIPE_FIELDS: 'accountStatusStripeFields',
  UPDATE_STRIPE_FIELDS_TOKEN: 'accountStatusUpdateStripeFieldsToken',
  UPDATE_STRIPE_FIELDS_REQUEST: 'accountStatusUpdateStripeFieldsRequest',
  UPDATE_DB_FIELDS: 'accountStatusUpdateDBFields',
  UPDATE_AWS_FIELDS: 'accountStatusUpdateAWSFields',
};

export const fieldNames = {
  NAME: 'accountName',
  EMAIL: 'accountEmail',
  EMAIL_ADDITIONAL: 'accountEmailAdditional',
  PHONE: 'accountPhone',
  NAME_ON_CARD: 'accountNameOnCard',
  PROMO_CODE: 'accountPromoCode',
  PROMO_CODE_VALID: 'accountPromoCodeValid',
  CUSTOMER: 'accountCustomer',
};
