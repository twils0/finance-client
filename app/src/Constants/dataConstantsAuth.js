export const actionTypes = {
  SET_AUTHENTICATED: 'DATA_AUTH_SET_AUTHENTICATED',
  SET_REDIRECT_URL: 'DATA_AUTH_SET_REDIRECT_URL',
  SET_AUTH_STATUS: 'DATA_AUTH_SET_AUTH_STATUS',
  SET_CODE_TYPE: 'DATA_AUTH_SET_CODE_TYPE',
  RESET_AUTH_STATE: 'DATA_AUTH_RESET_AUTH_STATE',
};

export const statusNames = {
  LOGIN: 'authStatusLogin',
  SIGN_OUT_DEVICES: 'authStatusSignOutDevices',
  LOGIN_MFA: 'authStatusLoginMFA',
  SIGN_UP: 'authStatusSignUp',
  VERIFY_PHONE: 'authStatusVerifyPhone',
  VERIFY_PHONE_CODE: 'authStatusVerifyPhoneCode',
  VERIFY_EMAIL: 'authStatusVerifyEmail',
  VERIFY_EMAIL_LINK: 'authStatusVerifyEmailLink',
  FORGOT_PASSWORD: 'authStatusForgotPassword',
  RESET_PASSWORD: 'authStatusResetPassword',
  CHANGE_PASSWORD: 'authStatusChangePassword',
  DELETE_ACCOUNT: 'authStatusDeleteAccount',
  LOGOUT: 'authStatusLogout',
};

export const codeTypeNames = {
  VERIFY_PHONE: 'verifyPhone',
  VERIFY_EMAIL: 'verifyEmail',
  VERIFY_EMAIL_ADDITIONAL: 'verifyEmailAdditional',
};
