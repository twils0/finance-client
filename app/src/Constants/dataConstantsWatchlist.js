export const actionTypes = {
  SET_WATCHLIST_STATUS: 'DATA_WATCHLIST_SECURITIES_STATUS',
  SET_SECURITIES_CURRENT: 'DATA_WATCHLIST_SET_SECURITIES_CURRENT',
  SET_SECURITIES_LIST: 'DATA_WATCHLIST_SET_SECURITIES_LIST',
  SET_SECURITIES_ALL: 'DATA_WATCHLIST_SET_SECURITIES_ALL',
  SET_SECURITY: 'DATA_WATCHLIST_SET_SECURITY',
  SET_SECURITY_DATA: 'DATA_WATCHLIST_SET_SECURITY_DATA',
  SET_SECURITY_SUBSCRIBE: 'DATA_WATCHLIST_SET_SECURITY_SUBSCRIBE',
  RESET_WATCHLIST_STATE: 'DATA_ACCOUNT_RESET_WATCHLIST_STATE',
};

export const codeTypeNames = {
  CONFIRM_PHONE: 'confirmPhone',
  CONFIRM_EMAIL: 'confirmEmail',
};

export const statusNames = {
  SECURITIES: 'watchlistStatusSecurities',
  GET_SECURITY_DATA: 'watchlistStatusGetSecurityData',
  UPDATE_SECURITIES: 'watchlistStatusUpdateSecurities',
};
