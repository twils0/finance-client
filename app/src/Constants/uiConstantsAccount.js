export const actionTypes = {
  SET_CURRENT_FORM: 'ACCOUNT_SET_CURRENT_FORM',
  SET_FORM_HEIGHT: 'ACCOUNT_SET_FORM_HEIGHT',
  SET_FORM_EDIT: 'ACCOUNT_SET_FORM_EDIT',
  SET_INPUT_VALUE_ERROR: 'ACCOUNT_SET_INPUT_VALUE_ERROR',
  SET_BUTTON_TEXT: 'ACCOUNT_SET_BUTTON_TEXT',
  SET_BUTTON_VISIBLE: 'ACCOUNT_SET_BUTTON_VISIBLE',
};

export const formNames = {
  CODE: 'codeForm',
  PROFILE: 'profileForm',
  BILLING: 'billingForm',
  CHANGE_PASSWORD: 'changePasswordForm',
  DELETE_ACCOUNT: 'deleteAccountForm',
};

export const buttonNames = {
  PROFILE: 'profileButton',
  BILLING: 'billingButton',
  CHANGE_PASSWORD: 'changePasswordButton',
  DELETE_ACCOUNT: 'deleteAccountButton',
  FIRST: 'firstButton',
  SECOND: 'secondButton',
};

export const buttonTexts = {
  RESEND: 'Resend',
  SUBMIT: 'Submit',
  EDIT: 'Edit',
  UPDATE: 'Update',
  CANCEL: 'Cancel',
  DELETE: 'Delete',
  CONFIRM: 'Confirm',
};

export const inputNames = {
  [formNames.CODE]: {
    CODE_PHONE: 'codeCodePhone',
  },
  [formNames.PROFILE]: {
    NAME: 'profileName',
    EMAIL: 'profileEmail',
    EMAIL_ADDITIONAL: 'profileEmailAdditional',
    PHONE: 'profilePhone',
  },
  [formNames.BILLING]: {
    NAME_ON_CARD: 'billingNameOnCard',
    PROMO_CODE: 'billingPromoCode',
    STRIPE: 'billingStripe',
  },
  [formNames.CHANGE_PASSWORD]: {
    OLD_PASSWORD: 'changePasswordOldPassword',
    NEW_PASSWORD: 'changePasswordNewPassword',
    NEW_PASSWORD2: 'changePasswordNewPassword2',
  },
  [formNames.DELETE_ACCOUNT]: {
    CONFIRM: 'deleteAccountConfirm',
  },
};
