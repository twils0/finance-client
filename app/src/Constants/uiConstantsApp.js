export const actionTypes = {
  SET_FONTS_STATUS: 'APP_SET_FONTS_STATUS',
  LOAD_FONTS: 'APP_LOAD_FONTS',
  SET_IMAGE_STATUS: 'APP_SET_IMAGE_STATUS',
  LOAD_IMAGE: 'APP_LOAD_IMAGE',
  SET_STRIPE_STATUS: 'APP_SET_STRIPE_STATUS',
  SET_STRIPE_ELEMENT_LOADED: 'APP_SET_STRIPE_ELEMENT_LOADED',
  LOAD_STRIPE: 'APP_FETCH_STRIPE',
  SET_STRIPE_OBJECT: 'APP_SET_STRIPE_OBJECT',
};

export const motionStatusTypes = {
  IDLE: 'IDLE',
  ACTIVE: 'ACTIVE',
};

export const imageNames = {
  EXAMPLE_HEADER: 'exampleHeader',
};

export const errorMessages = {
  NO_EMAIL: 'Please enter your email address.',
  INVALID_EMAIL: 'Please enter a valid email address.',
  NO_PASSWORD: 'Please enter your password.',
  NO_NEW_PASSWORD: 'Please enter your new password.',
  INVALID_PASSWORD:
    'Please enter a password containing at least 8 characters, one of the following special characters, $@!%*#?&, one letter, and one number.',
  NO_MATCH_PASSWORD: 'This password does not match your password above.',
  NO_NAME_ON_CARD: 'Please enter the name shown on your credit card.',
  NO_STRIPE: 'Please enter your credit card information.',
  INVALID_CARD_NUMBER: 'Please enter a valid credit card number.',
  INCOMPLETE_CARD_NUMBER: 'Please enter your full credit card number.',
  INVALID_CARD_EXPIRATION_DATE: 'Please enter a valid expiration date.',
  INVALID_CARD_CVC: 'Please enter a valid CVC.',
  INVALID_CARD_ZIP_CODE: 'Please enter a valid zip code.',
  NO_NAME: 'Please enter your name.',
  NO_PHONE: 'Please enter your mobile phone number.',
  INVALID_PHONE: 'Please enter a valid mobile phone number.',
  NO_VERIFICATION_CODE_PHONE: 'Please enter the verification code sent to your mobile phone.',
  NO_VERIFICATION_CODE_EMAIL: 'Please enter the verification code sent to your email.',
  NO_CONFIRMATION_CODE_PHONE: 'Please enter the confirmation code sent to your mobile phone.',
  INVALID_VERIFICATION_CODE:
    'Please verify you have entered the correct verification code or request a new code.',
  INVALID_CONFIRMATION_CODE:
    'Please verify you have entered the correct confirmation code or request a new code.',
  EXPIRED_VERIFICATION_CODE: 'This verification code has expired.',
  EXPIRED_CONFIRMATION_CODE: 'This confirmation code has expired.',
  EMAIL_IN_USE:
    'An account using this email already exists. Please login or use a different email.',
  EMAIL_IN_USE_INTERNAL:
    'An account using this email already exists. Please use a different email.',
  EMAIL_NOT_FOUND:
    'This email is not associated with an account. Please enter another email or sign up.',
  PASSWORD_NOT_FOUND: 'This password does not match our records.',
  PROMO_CODE_INVALID: 'Please enter a valid promo code.',
  PROMO_CODE_EXPIRED: 'This promo code is no longer valid.',
  RETRY_SIGN_UP: 'Please click "Cancel" below and login.',
  INVALID_DELETE_ACCOUNT: 'Please type "delete" to confirm you would like to delete your account',
  LIMIT_EXCEEDED:
    'You have reached your login attempt limit. Please allow some time and try again.',
  INTERNAL_ERROR:
    'Please try your request again. If unsuccessful on subsequent attempts, please contact customer service.',
};
