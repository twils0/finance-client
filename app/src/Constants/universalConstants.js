import stripeConfig from '../../../stripe.config.json';
import dbConfig from '../../../db.config.json';
import iexConfig from '../../../iex.config.json';

export const baseURLs = {
  STRIPE: stripeConfig.url,
  DB: dbConfig.url,
  IEX: iexConfig.urls.api,
  IEX_DEV: iexConfig.urls.dev,
  IEX_TERMS: iexConfig.urls.terms,
};

export const URLs = {
  CUSTOMERS: '/customers',
  USERS: '/users',
  EMAILS: '/emails',
  SECURITIES: '/securities',
  SECURITIES_SEARCH: '/securities/search',
  IEX_COMPANY: tickerCusip => `/stock/${tickerCusip}/company`,
  IEX_QUOTE: tickerCusip => `/stock/${tickerCusip}/quote`,
};

export const axiosConfig = {
  STRIPE: {
    baseURL: baseURLs.STRIPE,
    timeout: 30000,
  },
  DB: {
    baseURL: baseURLs.DB,
    timeout: 30000,
  },
  IEX: {
    baseURL: baseURLs.IEX,
    timeout: 30000,
  },
};

export const pathNames = {
  DEFAULT: '*',
  LOGIN: '/login',
  SIGN_UP: '/signup',
  CODE: '/code',
  CODE_VERIFY_EMAIL: '/code/:verificationId?',
  TERMS: '/terms',
  INTERNAL: '/internal',
  WATCHLIST: '/internal/watchlist',
  WATCHLIST_SECURITY_ID: '/internal/watchlist/:securityId?',
  ACCOUNT: '/internal/account',
};

export const requestStatusTypes = {
  IDLE: 'IDLE',
  LOADING: 'LOADING',
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR',
};
