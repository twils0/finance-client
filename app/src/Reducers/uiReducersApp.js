import { combineReducers } from 'redux';

import stripeConfig from '../../../stripe.config.json';
import { requestStatusTypes } from '../Constants/universalConstants';
import { actionTypes, imageNames } from '../Constants/uiConstantsApp';
import {
  formNames as formNamesSignUp,
  inputNames as inputNamesSignUp,
} from '../Constants/uiConstantsExternal';
import {
  formNames as formNamesAccount,
  inputNames as inputNamesAccount,
} from '../Constants/uiConstantsAccount';

import exampleHeader from '../Images/exampleHeader.png';

export const initialState = {
  fonts: {
    status: requestStatusTypes.IDLE,
  },
  images: {
    list: [...Object.values(imageNames)],
    [imageNames.EXAMPLE_HEADER]: {
      id: imageNames.EXAMPLE_HEADER,
      src: exampleHeader,
      status: requestStatusTypes.IDLE,
    },
  },
  stripe: {
    apiKey: stripeConfig.apiKey,
    status: requestStatusTypes.IDLE,
    elements: {
      [inputNamesSignUp[formNamesSignUp.CARD].STRIPE]: {
        id: inputNamesSignUp[formNamesSignUp.CARD].STRIPE,
        loaded: false,
      },
      [inputNamesAccount[formNamesAccount.BILLING].STRIPE]: {
        id: inputNamesAccount[formNamesAccount.BILLING].STRIPE,
        loaded: false,
      },
    },
    stripeObject: null,
  },
};

export const fonts = (state = initialState.fonts, action) => {
  switch (action.type) {
    case actionTypes.SET_FONTS_STATUS:
      return {
        ...state,
        ...action.payload,
      };
    default:
      return state;
  }
};

export const images = (state = initialState.images, action) => {
  switch (action.type) {
    case actionTypes.SET_IMAGE_STATUS:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    default:
      return state;
  }
};

export const stripe = (state = initialState.stripe, action) => {
  switch (action.type) {
    case actionTypes.SET_STRIPE_STATUS:
      return {
        ...state,
        ...action.payload,
      };
    case actionTypes.SET_STRIPE_ELEMENT_LOADED:
      return {
        ...state,
        elements: {
          ...state.elements,
          [action.payload.id]: {
            ...state.elements[action.payload.id],
            ...action.payload,
          },
        },
      };
    case actionTypes.SET_STRIPE_OBJECT:
      return {
        ...state,
        ...action.payload,
      };
    default:
      return state;
  }
};

const uiReducersApp = combineReducers({
  fonts,
  images,
  stripe,
});

export default uiReducersApp;
