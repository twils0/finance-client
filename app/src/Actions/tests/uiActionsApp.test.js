import configureMockStore from 'redux-mock-store';
import { initialState } from '../../Reducers/uiReducersApp';
import { requestStatusTypes } from '../../Constants/universalConstants';
import { actionTypes } from '../../Constants/uiConstantsApp';
import {
  setFontsStatus,
  setImageStatus,
  setStripeStatus,
  setStripeElementLoaded,
  setStripeObject,
} from '../uiActionsApp';

const middleware = [];
const mockStore = configureMockStore(middleware);

describe('uiActionsApp', () => {
  describe('setFontsStatus', () => {
    it("fails when not provided a payload with a 'status' key", () => {
      const store = mockStore(initialState);

      const payload = {};

      expect(() => store.dispatch(setFontsStatus(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = { status: requestStatusTypes.SUCCESS };
      const expectedActions = [
        {
          type: actionTypes.SET_FONTS_STATUS,
          payload,
        },
      ];

      store.dispatch(setFontsStatus(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });

  describe('setImageStatus', () => {
    it("fails when not provided a payload with a 'id' key", () => {
      const store = mockStore(initialState);

      const payload = { status: requestStatusTypes.SUCCESS };

      expect(() => store.dispatch(setImageStatus(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when not provided a payload with a 'status' key", () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId' };

      expect(() => store.dispatch(setImageStatus(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId', status: requestStatusTypes.SUCCESS };
      const expectedActions = [
        {
          type: actionTypes.SET_IMAGE_STATUS,
          payload,
        },
      ];

      store.dispatch(setImageStatus(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });

  describe('setStripeStatus', () => {
    it("fails when not provided a payload with a 'status' key", () => {
      const store = mockStore(initialState);

      const payload = {};

      expect(() => store.dispatch(setStripeStatus(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = { status: requestStatusTypes.SUCCESS };
      const expectedActions = [
        {
          type: actionTypes.SET_STRIPE_STATUS,
          payload,
        },
      ];

      store.dispatch(setStripeStatus(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });

  describe('setStripeElementLoaded', () => {
    it("fails when not provided a payload with a 'id' key", () => {
      const store = mockStore(initialState);

      const payload = { loaded: true };

      expect(() => store.dispatch(setImageStatus(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when not provided a payload with a 'loaded' key", () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId' };

      expect(() => store.dispatch(setImageStatus(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when provided a payload with a 'loaded' key that is not a boolean", () => {
      const store = mockStore(initialState);

      const payload = { loaded: 'loaded' };

      expect(() => store.dispatch(setStripeElementLoaded(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = { id: 'testId', loaded: true };
      const expectedActions = [
        {
          type: actionTypes.SET_STRIPE_ELEMENT_LOADED,
          payload,
        },
      ];

      store.dispatch(setStripeElementLoaded(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });

  describe('setStripeObject', () => {
    it("fails when not provided a payload with a 'stripeObject' key", () => {
      const store = mockStore(initialState);

      const payload = {};

      expect(() => store.dispatch(setStripeObject(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when provided a payload with a 'stripeObject' key that is not an object", () => {
      const store = mockStore(initialState);

      const payload = { stripeObject: [] };

      expect(() => store.dispatch(setStripeObject(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = { stripeObject: {} };
      const expectedActions = [
        {
          type: actionTypes.SET_STRIPE_OBJECT,
          payload,
        },
      ];

      store.dispatch(setStripeObject(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });
});
