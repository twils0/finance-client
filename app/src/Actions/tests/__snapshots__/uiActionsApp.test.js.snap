// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uiActionsApp setFontsStatus fails when not provided a payload with a 'status' key 1`] = `"Please enter a value for the 'status' key - {}"`;

exports[`uiActionsApp setImageStatus fails when not provided a payload with a 'id' key 1`] = `"Please enter a value for the 'id' key - {\\"status\\":\\"SUCCESS\\"}"`;

exports[`uiActionsApp setImageStatus fails when not provided a payload with a 'status' key 1`] = `"Please enter a value for the 'status' key - {\\"id\\":\\"testId\\"}"`;

exports[`uiActionsApp setStripeElementLoaded fails when not provided a payload with a 'id' key 1`] = `"Please enter a value for the 'id' key - {\\"loaded\\":true}"`;

exports[`uiActionsApp setStripeElementLoaded fails when not provided a payload with a 'loaded' key 1`] = `"Please enter a value for the 'status' key - {\\"id\\":\\"testId\\"}"`;

exports[`uiActionsApp setStripeElementLoaded fails when provided a payload with a 'loaded' key that is not a boolean 1`] = `"Please enter a value for the 'id' key - {\\"loaded\\":\\"loaded\\"}"`;

exports[`uiActionsApp setStripeObject fails when not provided a payload with a 'stripeObject' key 1`] = `"Please enter a value for the 'stripeObject' key - {}"`;

exports[`uiActionsApp setStripeObject fails when provided a payload with a 'stripeObject' key that is not an object 1`] = `"Please provide an object value for the 'stripeObject' key - {\\"stripeObject\\":[]}"`;

exports[`uiActionsApp setStripeStatus fails when not provided a payload with a 'status' key 1`] = `"Please enter a value for the 'status' key - {}"`;
